Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -104,6 +105,28 @@ static void webkit_user_content_manager_class_init(WebKitUserContentManagerClass
             g_cclosure_marshal_VOID__BOXED,
             G_TYPE_NONE, 1,
             WEBKIT_TYPE_JAVASCRIPT_RESULT);
+
+    /**
+     * WebKitUserContentManager::script-request-received:
+     * @manager: the #WebKitUserContentManager
+     * @user_script_message: the #WebKitUserScriptMessage holding the value received from the JavaScript world.
+     *
+     * This signal is emitted when JavaScript in a web view calls
+     * <code>window.webkit.messageHandlers.<name>.postMessage()</code>, after registering
+     * <code><name></code> using
+     * webkit_user_content_manager_register_script_message_handler_with_reply()
+     *
+     * Since: 2.38
+     */
+    signals[SCRIPT_REQUEST_RECEIVED] =
+        g_signal_new(
+            "script-request-received",
+            G_TYPE_FROM_CLASS(gObjectClass),
+            static_cast<GSignalFlags>(G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED),
+            0, nullptr, nullptr,
```

Comment:
The reason for having a `gboolean` return value from signal handlers is to allow returning `TRUE` from one to indicate that the script message was handled, and that no mode handlers should run. Reminder: there can be multiple handlers connected to the same signal, and only one can produce a result.

To ensure signal emission stops after a handler returns `TRUE`, we need to pass [g_signal_accumulator_true_handled](https://docs.gtk.org/gobject/func.signal_accumulator_true_handled.html) here:

```suggestion
            0, g_signal_accumulator_true_handled, nullptr,
```

A note in the signal documentation indicating that the first handler that returns `TRUE` results in no other handlers running would be welcome, too.